lib/onelogin/ruby-saml/utils.rb

```diff
@@ -139,23 +139,25 @@ def self.decrypt_data(encrypted_node, private_key)
       def self.retrieve_symmetric_key(encrypt_data, private_key)
         encrypted_key = REXML::XPath.first(
           encrypt_data,
-          "./ds:KeyInfo/xenc:EncryptedKey or \
-           //xenc:EncryptedKey[@Id=$id]",
-          { "ds" => DSIG, "xenc" => XENC },
-          { "id" =>  self.retrieve_symetric_key_reference(encrypt_data) }
+          "./ds:KeyInfo/xenc:EncryptedKey | ./KeyInfo/e:EncryptedKey | //xenc:EncryptedKey[@Id=$id]",
+          { "ds" => DSIG, "xenc" => XENC, "e" => XENC },
```

Comment:
why add "e" => XENC if already was defined "xenc" => XENC? It should be:

"./ds:KeyInfo/xenc:EncryptedKey | ./KeyInfo/xenc:EncryptedKey | //xenc:EncryptedKey[@Id=$id]",
{ "ds" => DSIG, "xenc" => XENC},